### PR TITLE
Update ebpf socket

### PR DESCRIPTION
**Pull request by thiagoftsm**
For we continue bringing eBPF programs it will be necessary to update our `libbpf` version, but socket was not loaded with the latest version, so I rewrite the code bringing some cleanup for it. With this code we are also removing blockers we had for network viewer.

The problem that was happening with the old code had relationship with tables not loaded and the way that functions were declared.

This PR was tested on:

| Linux Distribution | kernel version |
|-------------------|----------------|
| Slackware Current | 5.13.11  |
| Manjaro 21.1 | 5.10.56 |
| Slackware Current | 4.14.239 |
| CentOS 8.4 | 4.18.0-305 |
| CentOS 7.9.2009 | 3.10.0-1160 |

### CODE REVIEW

**Comment by thiagoftsm** [comment]
Nice, the code is compiled with `llvm 12.0.1`, but it fails with old versions.

File: kernel/socket_kern.c [hunk 1]
```diff
@@ -378,111 +296,104 @@ int netdata_inet_csk_accept(struct pt_regs* ctx)
...
+#if NETDATASEL < 2
     int ret = (int)PT_REGS_RC(ctx);
-
-    if (ret < 0) {
-        netdata_update_global(NETDATA_KEY_ERROR_TCP_SENDMSG, 1);
-    }
+    if (ret) {
+        libnetdata_update_global(&tbl_global_sock, NETDATA_KEY_ERROR_TCP_SENDMSG, 1);
+        sent = (size_t)ret;
+    } else
```
Comment (vlvkobal) [inline kernel/socket_kern.c:313]: I don't understand this part. You are incrementing the error counter here. But what if `ret > 0`?
Why `size = ret` if `ret < 0`?

Comment (vlvkobal) [reply kernel/socket_kern.c:313]: Shouldn't it be something like that:
```
    if (ret < 0) {
		libnetdata_update_global(&tbl_global_sock, NETDATA_KEY_ERROR_TCP_SENDMSG, 1);
        sent = 0;
    } else {
        sent = (size_t)ret;
    }
```

Comment (thiagoftsm) [reply kernel/socket_kern.c:313]: You are right, when I was rewriting I made a mistake here, your suggestion is a correct.

**Comment by UmanShahzad** [comment]
I'll check this out tonight.

**Review by vlvkobal — APPROVED** [review]

**Review by underhood — APPROVED** [review]
no comments other than things which would rightly be considered nit-picking

**Review by UmanShahzad — APPROVED** [review]

**Pull request merged**